build.yaml

```diff
@@ -52,6 +55,28 @@ steps:
    dependsOn:
     - default_ns
     - batch_pods_ns
+ - kind: deploy
+   name: deploy_ci_agent_sa
+   namespace:
+     valueFrom: batch_pods_ns.name
+   config: ci/ci-agent-sa.yaml
+   dependsOn:
+    - batch_pods_ns
+   scopes:
+    - test
+    - dev
+ - kind: deploy
+   name: deploy_ci_agent_role
```

Comment:
I'd merge these two deploys steps.  Makes things faster, and I find too much fragmentation in the yaml hard to follow.